Documentation/project-docs/platform-names.md

```diff
@@ -0,0 +1,39 @@
+# Platform Names
+
+Over the years, we've accumulated quite a few names, platforms, and components that can make it hard for folks to understand what we're referring to. This document has a list that will help to qualify what we mean by what.
+
+This will also list some aliases. As you'll see the aliases aren't always correct -- they are merely listed to help you find the better and less confusing terminology.
+
+## .NET Framework
+
+**Also referred to as**: Desktop, full framework, in-box framework
+
+This refers to the .NET Framework that shipped in 2002. It's the main framework folks target today and allows you to build a wide variety of applications, such as WinForms, WPF, ASP.NET, and command line tools.
+
+## .NET Core
```

Comment:
I think this section as a whole focuses too much on the historical context of the term, which made it a bit confusing for me to read.  Compared with the .NET Framework section, which is abundantly clear about what .NET Framework is and what it's used for.